From e76c5a9ad11c8604e2ca806993a64cfb35f2a9a8 Mon Sep 17 00:00:00 2001
From: William Horvath <william@horvath.blog>
Date: Fri, 27 Dec 2024 13:17:14 -0800
Subject: [PATCH] gdi32: Map SwapBuffers directly to opengl32.wglSwapBuffers.

Removes one level of indirection, not sure why it was done this way before.

---
 dlls/gdi32/gdi32.spec |  2 +-
 dlls/gdi32/opengl.c   | 15 ---------------
 2 files changed, 1 insertion(+), 16 deletions(-)

diff --git a/dlls/gdi32/gdi32.spec b/dlls/gdi32/gdi32.spec
index 5bde67ea371..a57dbad80d0 100644
--- a/dlls/gdi32/gdi32.spec
+++ b/dlls/gdi32/gdi32.spec
@@ -635,7 +635,7 @@
 @ stdcall StretchDIBits(long long long long long long long long long ptr ptr long long)
 @ stdcall StrokeAndFillPath(long)
 @ stdcall StrokePath(long)
-@ stdcall SwapBuffers(long)
+@ stdcall SwapBuffers(long) opengl32.wglSwapBuffers
 @ stdcall TextOutA(long long long str long)
 @ stdcall TextOutW(long long long wstr long)
 @ stdcall TranslateCharsetInfo(ptr ptr long)
diff --git a/dlls/gdi32/opengl.c b/dlls/gdi32/opengl.c
index 0fbd851f67e..d826c59f26d 100644
--- a/dlls/gdi32/opengl.c
+++ b/dlls/gdi32/opengl.c
@@ -37,7 +37,6 @@ static INT (WINAPI *wglChoosePixelFormat)(HDC,const PIXELFORMATDESCRIPTOR *);
 static INT (WINAPI *wglDescribePixelFormat)(HDC,INT,UINT,PIXELFORMATDESCRIPTOR*);
 static INT (WINAPI *wglGetPixelFormat)(HDC);
 static BOOL (WINAPI *wglSetPixelFormat)(HDC,INT,const PIXELFORMATDESCRIPTOR*);
-static BOOL (WINAPI *wglSwapBuffers)(HDC);
 
 /******************************************************************************
  *		ChoosePixelFormat (GDI32.@)
@@ -94,17 +93,3 @@ BOOL WINAPI SetPixelFormat( HDC hdc, INT fmt, const PIXELFORMATDESCRIPTOR *pfd )
     }
     return wglSetPixelFormat( hdc, fmt, pfd );
 }
-
-/******************************************************************************
- *		SwapBuffers (GDI32.@)
- */
-BOOL WINAPI SwapBuffers( HDC hdc )
-{
-    if (!wglSwapBuffers)
-    {
-        if (!opengl32) opengl32 = LoadLibraryW( L"opengl32.dll" );
-        if (!(wglSwapBuffers = (void *)GetProcAddress( opengl32, "wglSwapBuffers" )))
-            return FALSE;
-    }
-    return wglSwapBuffers( hdc );
-}
-- 
2.47.1

